### ISSUE TITLE
Proposal: Envoy Admission Control calculation to use latency statistics

### ISSUE THREAD

**Issue by marcusoyang** [envoyproxy/envoy#36415]
### tl:dr
Add latency statistics into the Admission Control filter success rate calculation

### Context
To make the Admission Control capability more effective in the context of slow dependency response times, the proposal is to add to the Admission Control filter [envoy/source/extensions/filters/http/admission_control at main · envoyproxy/envoy](https://github.com/envoyproxy/envoy/tree/main/source/extensions/filters/http/admission_control)  and add response latency statistics in its successRateThreshold calculation.

Currently, success rate is derived only from the HTTP response status codes of a dependency. This heuristic is effective when response latency is relatively fast, however in the situation when a dependency’s response latency is degraded, this would be useful.

**Comment by kyessenov** [comment]
CC @tonya11en

**Comment by tonya11en** [comment]
@marcusoyang I'm happy to explore this with you. Have you taken a look at [Adaptive Concurrency 
Control](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/adaptive_concurrency_filter)? That mechanism uses sampled latencies to make adjustments to the concurrency limit (essentially circuit breaking) thresholds.

I'm curious whether that meets your requirements and if not, why.

**Comment by ipoval** [comment]
@tonya11en , 👋 , thanks for a quick reply. @marcusoyang  have asked me about our experience with Admission Control and I've shared the potential latency-based success threshold for requests as one improvement item.  However I believe that it's not the most important item to pay attention. With my rollout experience across few services, it becomes clear that due to the fact that the counters are thread-based and not aggregated, the RPS rate per Envoy worker does not meet the requirement even in the cases of high RPS services (most of the time those services also have a lot of vCPU cores).

**Comment by ipoval** [comment]
@tonya11en  👋 , great to see you again! Sorry, things were a bit busy, but I'm glad that you are open to the idea of trying out the none-thread-local approach. It will probably take us a bit of time to come up with a proof of concept that we could share with you, and maybe in the meantime, I'll chat with @marcusoyang to see if we can close this issue to not distract you and once we have a spike on our end we could open a new thread 😃
